### DISCUSSION TITLE
When copilot goes hammock

### DISCUSSION THREAD

**Discussion by josias-r** [community/community#52594] [Copilot Conversations]
### Select Topic Area

Bug

### Body

I've been using copilot for a while now and I especially love it for writing documentation. During that use case, I had a few funny incidents where Copilot seems to be caught in a loop.

Here is the most recent example:

![Screenshot 2023-04-12 at 08 50 11](https://user-images.githubusercontent.com/11424820/231376119-d6d013f5-765f-4733-aa4f-01a3cb1b6732.png)

In this case Copilot ends up reaching for outer space. It's amusing to watch the sentences slowly progress to less and less sense until we end up at `cluster cluster cluster cluster` 🤣

Anyways. For one I wanted to report this issue, on the other hand I'm also curious if this is a known issue and if there is at least a theory on why this happened? I'm curious to learn a bit about the how and when GPT reaches a stop. 

It seems in my case I'm running into a hard stop. Why is this not always an issue, how is it "designed" to stop at a point where it makes sense. If that question makes sense :)

**Comment by Akash1134** [comment]
Hi @josias-r ,

Thanks for sharing your feedback, that means a lot. Regarding the issue you've highlighted above, as with most generative AI products, Copilot also makes its best guess based on the statistics and is subjected to making a few mistakes. 

When GitHub Copilot synthesizes code suggestions based on available public data, it can also synthesize code that contains some undesirable patterns. This is something we care a lot about at GitHub, and in recent years we’ve provided tools such as GitHub Actions, Dependabot, and CodeQL to open-source projects to help improve code quality. Of course, you should always use GitHub Copilot together with good testing and code review practices and security tools, as well as your own judgment.

I'd recommend reading these [FAQs](https://github.com/features/copilot#:~:text=Frequently%20asked%C2%A0questions) for more details about how Copilot suggests code and how you can leverage it at best. 🙂

**Reply by josias-r** [reply]
Thank you for your generic copy-paste answer, it does obviously not answer my question.
But that's alright, I don't really need an answer 😉 because I was only curious after all about some technical details on LLM like GPT on how the stopping mechanism works (if you can call it that).

So if someone with some techy insights into my question stumbles upon this discussion at any time in the future, I would love if you could share some knowledge.

Otherwise consider this dicussion as closed :)
